File: typescript/pangram/pangram.ts
const NUMBER_OF_LETTERS_IN_ALPHABET = 26;
class Pangram {
  constructor(private readonly text: string) {}

  isPangram(): boolean {
    return this.countUniqueLetters() === NUMBER_OF_LETTERS_IN_ALPHABET;
  }

  private countUniqueLetters(): number {
    return new Set(this.text.toLowerCase().match(/[a-z]/g)).size;
  }
}

export default Pangram;

File: go/pangram/pangram.go
// Package pangram exposes functions to analyze pangrams.
package pangram

import (
	"strings"
)

// IsPangram determines whether a text is a pangram.
func IsPangram(text string) bool {
	letters := map[rune]bool{}

	for _, ch := range strings.ToLower(text) {
		letters[ch] = true
	}

	for ch := 'a'; ch <= 'z'; ch++ {
		if !letters[ch] {
			return false
		}
	}

	return true
}
